libs/open_dashboard_backend/data_models.h

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2020 twyleg
+#pragma once
+
+#include <QMetaProperty>
+#include <QObject>
+
+#include <QUrl>
+
+#define AddAttribute(datatype, name, funcName, initial_value) \
+private: \
+	datatype name = initial_value; \
```

Comment:
what is this? Why not just type it out? Or is this standard practice for this sort of thing